In [3]:
import data_prepare
import numpy as np
from sklearn import svm
from sklearn.model_selection import GridSearchCV

In [4]:
path = 'E:\PersonalFiles\杂项\Dissertation\Data/diabetes_data.csv'
x_train, x_test, y_train, y_test = data_prepare.split_data(path)

In [5]:
parameters = {'kernel':('linear', 'rbf', 'poly', 'sigmoid'), 
              'C':np.linspace(0.01, 10, 100)}
svc = svm.SVC()
clf = GridSearchCV(svc, parameters, n_jobs=-1, scoring='accuracy', cv=5)
clf.fit(x_train, y_train)

data_prepare.print_best_score(clf, parameters)

Best score: 0.981
Best parameters set:
	C: 2.3309090909090906
	kernel: 'rbf'


In [8]:
print('acc on training set: ', clf.score(x_test, y_test))

acc on training set:  0.9487179487179487


结论：在SVM中，使用rbf作为核，正则化参数C选择2.33时，取到最优解，准确率在训练集上为98.1%(5折交叉检验)，在测试集上为94.9%